### PR TITLE
Remote runtime configuration with DAPR concept

### PR DESCRIPTION
**Pull request by kfurgol**
**Description**

Changes proposed in this pull request:

- Added a document describing the idea of remote runtime configuration using DAPR

**Related issue(s)**
#11

### CODE REVIEW

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+## Overview
+
+This document describes the concept of passing a configuration to runtimes from the central point with the usage of [DAPR](https://dapr.io).
+
+![Concept image](assets/concept.png?raw=true "Concept")
+
+## Idea
+
+The main idea is to have a component in Kyma Control Plane (Runtime Director on image), which would be a point of communication with Runtimes through Agents and provide a configuration for them. The configuration would be for example an URL and credentials to some external service (e.g. logging service). After each runtime fetch the new configuration, the DAPR bindings should be updated and populated to the DAPR sidecars.
```
Comment (crabtree) [inline docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:11]: I have an impression from that sentence that the main reason of having the Runtime Director component is to have a possibility to deliver configuration to the runtimes. I would rephrase this to more underline the intention of having the Runtime Director for runtime management. Then in some details we can focus on the configuration part. 

I'm also missing the big picture describing the architecture presented on the concept image.

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## Example
```
Comment (crabtree) [inline docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:13]: This example fits more to the actual POC documentation, describing the actual use-case to implement.

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Passing configuration to runtimes concept
```
[the remark this answers is not in the document]
Comment (kfurgol) [reply docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md]: What do you think about `Remote runtime management concept`?

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+## Overview
+
+This document describes the concept of passing a configuration to runtimes from the central point with the usage of [DAPR](https://dapr.io).
+
+![Concept image](assets/concept.png?raw=true "Concept")
```
[the remark this answers is not in the document]
Comment (kfurgol) [reply docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md]: Well, I thought of this document as a description where the concept of Runtime management lays in the big picture of Kyma environment. I'd go with more specific diagrams on the POC doc. What do you thing about that?

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+The diagram above shows the future architecture of Kyma environment.
+There are two central places, Compass and Kyma Control Plane along with runtimes which will communicate with them.
+
+### Compass
+It is responsible for connecting applications to runtimes, along with controlling and monitoring them.
+The System Broker will be responsible for fetching data from different types of runtimes (e.g. Cloud Foundry or customer Kubernetes cluster with service catalog installed on it).  
+
+### Kyma Control Plane
+It is responsible for provisioning new Kyma runtimes and providing initial configurations for them.
```
Comment (crabtree) [inline docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:19]:
```suggestion
The Kyma Control Plane is responsible for managing the Kyma instances. It allows to provision Kyma Runtimes as well as it is a central place where the Kyma instances can fetch their configuration from. 
```

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+The diagram above shows the future architecture of Kyma environment.
+There are two central places, Compass and Kyma Control Plane along with runtimes which will communicate with them.
+
+### Compass
+It is responsible for connecting applications to runtimes, along with controlling and monitoring them.
+The System Broker will be responsible for fetching data from different types of runtimes (e.g. Cloud Foundry or customer Kubernetes cluster with service catalog installed on it).  
+
+### Kyma Control Plane
+It is responsible for provisioning new Kyma runtimes and providing initial configurations for them. 
+The Runtime Director will be a service that will be responsible for managing and controlling runtimes through runtime agents.
```
Comment (crabtree) [inline docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:20]:
```suggestion
The Runtime Director exposes the API for the Kyma Runtimes configuration. 
```

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+### Runtimes
+The diagram presents some possible runtimes, which will be supported in the future. It includes Cloud Foundry runtime, Customer own cluster with Service Catalog installed, Kyma Managed Runtime, and Kyma Standalone Runtime.
```
Comment (crabtree) [inline docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:23]:
```suggestion
The diagram shows possible runtimes, which will be supported. It includes Cloud Foundry runtime, Customer own cluster with Service Catalog installed, Kyma Managed Runtime, and Kyma Standalone Runtime. 
```

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+It is responsible for connecting applications to runtimes, along with controlling and monitoring them.
+The System Broker will be responsible for fetching data from different types of runtimes (e.g. Cloud Foundry or customer Kubernetes cluster with service catalog installed on it).  
+
+### Kyma Control Plane
+It is responsible for provisioning new Kyma runtimes and providing initial configurations for them. 
+The Runtime Director will be a service that will be responsible for managing and controlling runtimes through runtime agents.
+
+### Runtimes
+The diagram presents some possible runtimes, which will be supported in the future. It includes Cloud Foundry runtime, Customer own cluster with Service Catalog installed, Kyma Managed Runtime, and Kyma Standalone Runtime. 
+Each of them beside Kyma Standalone will have some implementation of Agent, that will communicate with Control Plane.
```
Comment (crabtree) [inline docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:24]:
```suggestion
Each supported Kyma instance has a dedicated agent component that is responsible for handling the communication with the Kyma Central Plane. The agent component fetches the configuration from the Runtime Director and applies it to the desired environment.
```

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+### Kyma Control Plane
+The Kyma Control Plane is responsible for managing the Kyma instances. It allows to provision Kyma Runtimes as well as it is a central place where the Kyma instances can fetch their configuration from. 
+The Runtime Director exposes the API for the Kyma Runtimes configuration. 
+
+### Runtimes
+The diagram shows possible runtimes, which will be supported. It includes Cloud Foundry runtime, Customer own cluster with Service Catalog installed, Kyma Managed Runtime, and Kyma Standalone Runtime. 
+Each supported Kyma instance has a dedicated agent component that is responsible for handling the communication with the Kyma Central Plane. The agent component fetches the configuration from the Runtime Director and applies it to the desired environment.
+
+## Kyma Runtime configuration from the Control Plane
+The main idea is to have a component in Kyma Control Plane (Runtime Director on image), which would be a point of communication with Runtimes through Agents to manage them. It would allow also to configure them from one central point.
```
Comment (klaudiagrz) [inline docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:27]:
```suggestion
The main idea is to have a component in Kyma Control Plane that will be a point of communication for Agents to manage the Runtimes. This component would be the image of Runtime Director. It would also allow for configuring Runtimes from one central place.
```

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+The configuration could be for example an URL and credentials to some external service (e.g. logging service) which is used by some of the registered Runtimes. After each runtime fetches the new configuration, the Dapr bindings would be updated and populated to the Dapr sidecars.
```
Comment (klaudiagrz) [inline docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:29]:
```suggestion
For example, the configuration could be passed as an URL with credentials to some external service, such as logging service which is used by some of the registered Runtimes. After each Runtime fetches the new configuration, the Dapr bindings would be updated and populated to the Dapr sidecars.
```

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+### Kyma Control Plane
+The Kyma Control Plane is responsible for managing the Kyma instances. It allows to provision Kyma Runtimes as well as it is a central place where the Kyma instances can fetch their configuration from. 
+The Runtime Director exposes the API for the Kyma Runtimes configuration. 
+
+### Runtimes
+The diagram shows possible runtimes, which will be supported. It includes Cloud Foundry runtime, Customer own cluster with Service Catalog installed, Kyma Managed Runtime, and Kyma Standalone Runtime. 
+Each supported Kyma instance has a dedicated agent component that is responsible for handling the communication with the Kyma Central Plane. The agent component fetches the configuration from the Runtime Director and applies it to the desired environment.
+
+## Kyma Runtime configuration from the Control Plane
```
[the remark this answers is not in the document]
Comment (kfurgol) [reply docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:29]: Actually this is true, except for the Dapr part. It is not said that every runtime would be using Dapr even if we decided to go with it for Managed Kyma Runtimes. I will rephrase this sentence

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+## Architecture
+
+![Concept image](assets/concept.png?raw=true "Concept")
+
+The diagram above shows the future architecture of Kyma environment.
+There are two central places, Compass and Kyma Control Plane along with runtimes which will communicate with them.
+
+### Compass
+It is responsible for connecting applications to runtimes, along with controlling and monitoring them.
+The System Broker will be responsible for fetching data from different types of runtimes (e.g. Cloud Foundry or customer Kubernetes cluster with service catalog installed on it).
```
[the remark this answers is not in the document]
Comment (kfurgol) [reply docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:15]: Good idea, I will change that sentence

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+- Cloud Foundry 
+- customer own Kubernetes cluster with Service Catalog installed
+- managed Kyma Runtime
+- standalone Kyma Runtime
+Each supported Kyma instance has a dedicated Agent component that is responsible for handling the communication with the Kyma Central Plane. The Agent component fetches the configuration from the Runtime Director and applies it to the desired environment.
+
+## Managing Runtimes configuration in Kyma Control Plane
+The main idea is to have a component in Kyma Control Plane that will be a point of communication for Agents to manage the Runtimes. This component would be the image of Runtime Director. It would also allow for configuring Runtimes from one central place.
+
+For example, the configuration could be passed as an URL with credentials to some external service, such as logging service which is used by some of the registered Runtimes. In case of Managed Kyma Runtimes, after each Runtime fetches the new configuration, the Dapr bindings would be updated and populated to the Dapr sidecars.
```
Comment (klaudiagrz) [inline docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:32]:
```suggestion
For example, the configuration could be passed as an URL with credentials to some external service, such as logging service which is used by some of the registered Runtimes. In the case of managed Kyma Runtimes, after each Runtime fetches the new configuration, the Dapr bindings would be updated and populated to the Dapr sidecars.
```

File: docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md [hunk 14]
```diff
@@ -0,0 +1,32 @@
+# Remote Runtime management concept
+
+
```
Comment (klaudiagrz) [inline docs/internal/investigations/dapr-remote-runtime-configuration/configuration_concept.md:3]:
```suggestion
```
